File: sdks/web/test/branching.test.mjs
import assert from 'node:assert/strict';
import test from 'node:test';
import {SurveyFlow,pageRoute,routedAnswers} from '../dist/branching.js';
import fixture from '../../../contracts/branching-survey.example.json' with {type:'json'};

test('forward branches, Back history, progress and answer changes stay coherent',()=>{
  const schema={questions:fixture.questions,pages:fixture.pages};const flow=new SurveyFlow(schema);
  flow.setAnswer('return','no');assert.equal(flow.next(),true);assert.equal(flow.page.id,'recovery');assert.deepEqual(flow.progress,{current:3,total:4,visited:2});
  flow.setAnswer('problem','Long wait');assert.equal(flow.next(),true);assert.equal(flow.page.id,'contact');assert.equal(flow.back(),true);assert.equal(flow.page.id,'recovery');
  flow.back();flow.setAnswer('return','yes');assert.equal(flow.next(),true);assert.equal(flow.page.id,'praise');assert.ok(!('problem' in flow.answers));
});
test('answers from impossible skipped routes are removed',()=>{const schema={questions:fixture.questions,pages:fixture.pages};const answers={return:'no',highlight:'stale',problem:'Long wait'};assert.deepEqual(pageRoute(schema,answers),[0,2,3]);assert.deepEqual(routedAnswers(schema,answers),{return:'no',problem:'Long wait'});});

File: sdks/flutter/test/branching_test.dart
import 'dart:convert';
import 'dart:io';
import 'package:flutter_test/flutter_test.dart';
import 'package:likerts/branching.dart';
import 'package:likerts/likerts.dart';
void main(){test('branch navigation progress Back and answer changes are deterministic',(){final fixture=jsonDecode(File('../../contracts/branching-survey.example.json').readAsStringSync());final collection=Collection.fromJson({'id':'c','surveyId':'s','version':1,'placement':'p','schema':{'schemaVersion':4,...fixture}});final flow=SurveyFlow(collection);flow.setAnswer('return','no');expect(flow.next(),true);expect(flow.page.id,'recovery');expect(flow.progress,const SurveyProgress(3,4,2));flow.setAnswer('problem','Long wait');expect(flow.next(),true);expect(flow.page.id,'contact');expect(flow.back(),true);expect(flow.page.id,'recovery');expect(flow.back(),true);flow.setAnswer('return','yes');expect(flow.next(),true);expect(flow.page.id,'praise');expect(flow.answers.containsKey('problem'),false);expect(pageRoute(collection,{'return':'no'}),[0,2,3]);expect(routedAnswers(collection,{'return':'no','highlight':'stale'}),{'return':'no'});});}
